src/utils.js

```diff
@@ -509,13 +478,18 @@ export const isEncodingType = encodingType =>
   !!(
     encodingType &&
     typeof encodingType === 'string' &&
-    ['ascii', 'utf8', 'utf16le', 'ucs2', 'base64', 'binary', 'hex'].includes(
-      encodingType
-    )
+    ['ascii', 'utf8', 'utf16le', 'ucs2', 'base64', 'binary', 'hex'].includes(encodingType)
   );
 
-export const isEmptyString = str =>
-  !!(!str || (typeof str === 'string' && str.length === 0));
+export const isEmptyString = str => !!(!str || (typeof str === 'string' && str.length === 0));
 
-export const isValidHttpRequestBody = reqBody =>
-  !!(reqBody && (typeof reqBody === 'string' || reqBody instanceof Buffer));
+export const runOneTimeWrapper = (func, context) => {
+  let done = false;
+  return (...args) => {
+    if (!done) {
+      const result = func.apply(context || this, args);
```

Comment:
MMMMM, And how you want to indicate the user the function not really runs? it's not like a cache. it's more like a limitation. Thinking out loud: I don't want to hide this from the user (it will hide bugs if we will hide this)